tools/ci_build/github/azure-pipelines/win-xnn-ci-pipeline.yml

```diff
@@ -0,0 +1,64 @@
+##### start trigger Don't edit it manually, Please do edit set-trigger-rules.py ####
+### please do rerun set-trigger-rules.py ###
+trigger:
+  branches:
+    include:
+    - main
+    - rel-*
+  paths:
+    exclude:
+    - docs/**
+    - README.md
+    - CONTRIBUTING.md
+    - BUILD.md
+    - 'js/web'
+    - 'onnxruntime/core/providers/js'
+pr:
+  branches:
+    include:
+    - main
+    - rel-*
+  paths:
+    exclude:
+    - docs/**
+    - README.md
+    - CONTRIBUTING.md
+    - BUILD.md
+    - 'js/web'
+    - 'onnxruntime/core/providers/js'
+#### end trigger ####
+
+jobs:
+- job: 'build'
+  pool: 'onnxruntime-Win-CPU-2022'
+  variables:
+    MsbuildArguments: '-detailedsummary -maxcpucount -consoleloggerparameters:PerformanceSummary'
+    buildArch: x64
+    BuildConfig: 'RelWithDebInfo'
+    TODAY: $[format('{0:dd}{0:MM}{0:yyyy}', pipeline.startTime)]
+  timeoutInMinutes: 120
+  workspace:
+    clean: all
+  steps:
+
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.8'
+      addToPath: true
+      architecture: $(buildArch)
+
+  - template: templates/jobs/win-ci-build-steps.yml
+    parameters:
+      WithCache: True
+      Today: $(TODAY)
+      AdditionalKey: "win-xnn | $(BuildConfig)"
+      BuildPyArguments: '--config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --compile_no_warning_as_error --update --cmake_generator "Visual Studio 17 2022" --use_xnnpack --parallel --use_binskim_compliant_compile_flags'
```

Comment:
Does this also need `--build`? Otherwise the `--update` is going to generate all the cmake files but nothing will actually run the build.

Reply:
In fact, it splits the building into 2 steps, generate cmake and building sln in the template https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/azure-pipelines/templates/jobs/win-ci-build-steps.yml
It may use ccache to accelerate compiling and msbuild parameters may be changed.
To hide the details to caller, I don't add --build here.

Reply:
There's a test run https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1477478&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774

Reply:
ah ok. didn't realize the build was via msbuild not build.py